kv/kvserver: TestStrictGCEnforcement failed

kv/kvserver.TestStrictGCEnforcement [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14450104?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14450104?buildTab=artifacts#/) on master @ [4deb9e3b77f38396a58bb8186fe51bb01cf9f007](https://github.com/cockroachdb/cockroach/commits/4deb9e3b77f38396a58bb8186fe51bb01cf9f007):


```
=== RUN   TestStrictGCEnforcement
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/983c7b191abe1bb6a7f2826676d0ed84/logTestStrictGCEnforcement2813583560
    test_log_scope.go:81: use -show-logs to present logs inline
    client_replica_test.go:4431: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/983c7b191abe1bb6a7f2826676d0ed84/logTestStrictGCEnforcement2813583560
--- FAIL: TestStrictGCEnforcement (40.21s)
=== RUN   TestStrictGCEnforcement/zone_config_changes_are_respected
    client_replica_test.go:4314: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_replica_test.go:4146
        	            				github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_replica_test.go:4314
        	Error:      	Expect "<nil>" to match "1710795553.186268977,0 must be after replica GC threshold "
        	Test:       	TestStrictGCEnforcement/zone_config_changes_are_respected
    --- FAIL: TestStrictGCEnforcement/zone_config_changes_are_respected (14.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStrictGCEnforcement.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-36817

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -4076,6 +4076,13 @@ func TestStrictGCEnforcement(t *testing.T) {
 		ReplicationMode: base.ReplicationManual,
 		ServerArgs: base.TestServerArgs{
 			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					// We don't strictly enforce the GC TTL if the protected timestamp
+					// state cached on the replica is older than the lease's start time.
+					// We disable the lease queue to prevent flakes right around lease
+					// transfers. See getImpliedGCThresholdRLocked for more details.
+					DisableLeaseQueue: true,
+				},
 				SpanConfig: &spanconfig.TestingKnobs{
 					KVSubscriberRangeFeedKnobs: &rangefeedcache.TestingKnobs{
 						OnTimestampAdvance: func(timestamp hlc.Timestamp) {

